text/nodejs/0014-yarn-berry.md

```diff
@@ -0,0 +1,280 @@
+to do so. For those who opt out of "Plug 'n Play", Berry still supports using
+`node_modules`.
+
+The buildpack should run `yarn install` in the following cases:
+
+- If an application uses `node_modules`, which may be determined by the
+  presence of a `node_modules` directory or through `.yarnrc.yml` configuration.
+
+- If the `yarn.lock` has changed when using `node_modules`.
+
+- If there is no local cache (`.yarn/cache` by default, but can be configured) in
+  the working directory, whether `.pnp.cjs` exists or not.
+
+- If there is no `.pnp.cjs` in the working directory, even if a local cache exists.
+
+The buildpack should NOT run `yarn install` when:
+
+- Both a `.pnp.cjs` file and a local cache are present in the working
+  directory. The buildpack should assume that these two artifacts indicate a
+  desire to leverage Plug'n'Play and should not need to run `yarn install`
+  whether or not the `yarn.lock` has changed between builds.
+
+
+The buildpack will store the Yarn cache in a layer and configure Yarn to
+reference this layer, ideally via the `YARN_CACHE_FOLDER` environment variable.
+For Berry projects using `node_modules`, the behaviour will largely mimic the
+existing Yarn Install buildpack, with accommodations for the CLI changes.
+
+The buildpack will generate a filesystem-based SBoM as `yarn-install` does currently.
+
+### Layer Reuse 
+
+The Yarn Install buildpack currently uses a checksum of the output from `yarn
+config list` and the contents of `yarn.lock` to decide whether to reuse package layers.
+`yarn config list` has been removed in Berry, but the new `yarn info` command, which
+provides information about the project's direct and transitive dependencies,
+should work well as a replacement.
+
+### `--frozen-lockfile`
+
+The existing Yarn Install buildpack passes the `--frozen-lockfile` argument to
+every `yarn install` call, which will not update the `yarn.lock` file, and will
+fail if the `yarn.lock` file needs to be updated. As such, developers must run
+`yarn install` locally in order to update their app's `yarn.lock` file.
+`--frozen-lockfile` has been deprecated in favour of `--immutable`, which this
+buildpack will utilize.
+
+
+### Offline builds
+
+Previously, the `yarn-install` buildpack relied on `yarn-offline-mirror` along
+with the `--offline` flag to facilitate offline builds. This workflow is no
+longer supported and is indeed unnecessary since `.yarn/cache` stores archives
+of all packages now anyway.
+
+### Running Scripts
+
+Berry has [changed the way pre- & post- scripts are
+invoked](https://yarnpkg.com/getting-started/migration#explicitly-call-the-pre-and-post-scripts).
+
+The build logic of `yarn-start` will need to be modified to accommodate this
+change.
+
+### `yarn node` vs `node` 
+
+All `node` commands run via the shell should be run with `yarn node` instead. The default
+start command in the Yarn Start buildpack may need to be modified.
```

Comment:
Can we elaborate on any changes needed to the yarn-start or node-start (included in the berry group above) buildpacks? Seems like there could be some gotchas here with detection criteria and cache vs modules configurations.